packages/journey-manager/src/ui.ts

```diff
@@ -0,0 +1,380 @@
+/* eslint-disable accessor-pairs */
+import { type Client } from "@nlxai/voice-compass";
+
+/**
+ * Theme colors
+ */
+export interface ThemeColors {
+  /**
+   * Primary color
+   */
+  primary: string;
+  /**
+   * Primary color on hover
+   */
+  primaryHover: string;
+}
+
+/**
+ * Visual theme for the UI
+ */
+export interface Theme {
+  /**
+   * UI colors
+   */
+  colors: ThemeColors;
+  /**
+   * Font family
+   */
+  fontFamily: string;
+}
+
+/**
+ * Deep partial variant of the UI theme, input by the library user
+ */
+export interface PartialTheme {
+  /**
+   * UI colors
+   */
+  colors?: Partial<ThemeColors>;
+  /**
+   * Font family
+   */
+  fontFamily?: string;
+}
+
+/**
+ * Full UI configuration
+ */
+export interface UiConfig {
+  /**
+   * Drawer title
+   */
+  title: string;
+  /**
+   * Drawer subtitle
+   */
+  subtitle: string;
+  /**
+   * UI theme
+   */
+  theme?: PartialTheme;
+  /**
+   * Escalation step ID
+   */
+  escalationStep?: string;
+  /**
+   * End step ID
+   */
+  endStep?: string;
+}
+
+const defaultTheme: Theme = {
+  colors: {
+    primary: "",
+    primaryHover: "",
+  },
+  fontFamily: "sans-serif",
+};
+
+const mergeWithDefault = (partial?: PartialTheme): Theme => ({
+  ...defaultTheme,
+  ...(partial ?? {}),
+  colors: {
+    ...defaultTheme.colors,
+    ...(partial?.colors ?? {}),
+  },
+});
+
+const styles = (theme: Theme): string => `
```

Comment:
I'm kind of curious why use javascript for the theme-ing rather than CSS custom properties?